### PR TITLE
Nginx 1.9.5+ support

### LINKED ISSUE #2059: nginx/1.9.13 causes dokku install failure
Description of problem:
Changing the NGINX default install to 1.9.x via apt causes this error:
```shell
Adding user dokku to group adm
dpkg: error processing package dokku (--configure):
 subprocess installed post-installation script returned error exit status 1
```

Output of the following commands

- `uname -a`:
```text
Linux ubuntu-1gb-nyc3-01 3.13.0-79-generic #123-Ubuntu SMP Fri Feb 19 14:27:58 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
```
- `free -m`:
```text
total       used       free     shared    buffers     cached
Mem:           994        138        855          0         36         57
-/+ buffers/cache:         43        950
Swap:            0          0          0
```
- `docker version`:
```text
Client:
 Version:      1.10.3
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   20f81dd
 Built:        Thu Mar 10 15:54:52 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.10.3
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   20f81dd
 Built:        Thu Mar 10 15:54:52 2016
 OS/Arch:      linux/amd64
```
- `docker -D info`:
```text
Client:
 Version:      1.10.3
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   20f81dd
 Built:        Thu Mar 10 15:54:52 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.10.3
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   20f81dd
 Built:        Thu Mar 10 15:54:52 2016
 OS/Arch:      linux/amd64
root@ubuntu-1gb-nyc3-01:~# docker -D info
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 3
Server Version: 1.10.3
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 6
 Dirperm1 Supported: false
Execution Driver: native-0.2
Logging Driver: json-file
Plugins:
 Volume: local
 Network: null host bridge
Kernel Version: 3.13.0-79-generic
Operating System: Ubuntu 14.04.4 LTS
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 994 MiB
Name: ubuntu-1gb-nyc3-01
ID: OZGU:QAPI:4SCG:2VCF:AYBO:RRZV:PB7J:3OSU:EJF2:JLIC:SWSI:2MXL
WARNING: No swap limit support
```
- `sigil -v`: 0.4.0
- `docker run -ti gliderlabs/herokuish:latest herokuish version`:
```text
herokuish: 0.3.9
buildpacks:
  heroku-buildpack-clojure   v75
  heroku-buildpack-erlang    fa17af9
  heroku-buildpack-go        v31
  heroku-buildpack-gradle    v17
  heroku-buildpack-grails    v20
  heroku-buildpack-java      v44
  heroku-buildpack-multi     v1.0.0
  heroku-buildpack-nodejs    v88
  heroku-buildpack-php       v95
  heroku-buildpack-play      v26
  heroku-buildpack-python    v77
  heroku-buildpack-ruby      v145
  heroku-buildpack-scala     v66
  buildpack-nginx            v5
```
- `dokku version`:
```text
cat: /home/dokku/VERSION: No such file or directory
Unable to determine dokku's version
```
- `dokku plugin`:
```text
plugn: dev
  00_dokku-standard    0.5.3 enabled    dokku core standard plugin
  20_events            0.5.3 enabled    dokku core events logging plugin
  apps                 0.5.3 enabled    dokku core apps plugin
  build-env            0.5.3 enabled    dokku core build-env plugin
  certs                0.5.3 enabled    dokku core certificate management plugin
  checks               0.5.3 enabled    dokku core checks plugin
  common               0.5.3 enabled    dokku core common plugin
  config               0.5.3 enabled    dokku core config plugin
  docker-options       0.5.3 enabled    dokku core docker-options plugin
  domains              0.5.3 enabled    dokku core domains plugin
  enter                0.5.3 enabled    dokku core enter plugin
  git                  0.5.3 enabled    dokku core git plugin
  logs                 0.5.3 enabled    dokku core logs plugin
  named-containers     0.5.3 enabled    dokku core named containers plugin
  nginx-vhosts         0.5.3 enabled    dokku core nginx-vhosts plugin
  plugin               0.5.3 enabled    dokku core plugin plugin
  proxy                0.5.3 enabled    dokku core proxy plugin
  ps                   0.5.3 enabled    dokku core ps plugin
  shell                0.5.3 enabled    dokku core shell plugin
  storage              0.5.3 enabled    dokku core storage plugin
  tags                 0.5.3 enabled    dokku core tags plugin
  tar                  0.5.3 enabled    dokku core tar plugin
```

Environment details (AWS, VirtualBox, physical, etc.):
DigitalOcean Droplet (NYC3)
  - Ubuntu 14.04.4 x64
  - 1 GB Memory
  - 30 GB Disk

How was dokku installed?:
Instructions from http://dokku.viewdocs.io/dokku/
```shell
# only works on debian systems!
$ wget https://raw.githubusercontent.com/dokku/dokku/v0.5.3/bootstrap.sh
$ sudo DOKKU_TAG=v0.5.3 bash bootstrap.sh
# go to your server's IP and follow the web installer
```

How reproducible: Very reproducible.

Steps to Reproduce:
```
apt-get update && apt-get dist-upgrade -y && reboot
curl http://nginx.org/keys/nginx_signing.key | apt-key add -
echo "deb http://nginx.org/packages/mainline/ubuntu/ trusty nginx" | tee --append /etc/apt/sources.list.d/nginx_org_packages_mainline_ubuntu.list
wget https://raw.githubusercontent.com/dokku/dokku/v0.5.3/bootstrap.sh
DOKKU_TAG=v0.5.3 TRACE=1 bash bootstrap.sh
```

Actual Results:
```text
-----> Migrating zero downtime env variables. The following variables have been deprecated
=====> DOKKU_SKIP_ALL_CHECKS DOKKU_SKIP_DEFAULT_CHECKS
=====> Please use dokku checks:[disable|enable] <app> to control zero downtime functionality
+ for plugin in '$PLUGIN_PATH/enabled/*'
+++ basename /var/lib/dokku/core-plugins/enabled/common
++ config-export common
++ declare 'desc=Export plugin configuration'
++ declare plugin=common
++ toml-export common
++ /usr/local/bin/plugn ::: toml-export common
++ cat /var/lib/dokku/core-plugins/config.toml
+ eval ''
+ [[ -x /var/lib/dokku/core-plugins/enabled/common/install ]]
+ for plugin in '$PLUGIN_PATH/enabled/*'
+++ basename /var/lib/dokku/core-plugins/enabled/config
++ config-export config
++ declare 'desc=Export plugin configuration'
++ declare plugin=config
++ toml-export config
++ /usr/local/bin/plugn ::: toml-export config
++ cat /var/lib/dokku/core-plugins/config.toml
+ eval ''
+ [[ -x /var/lib/dokku/core-plugins/enabled/config/install ]]
+ for plugin in '$PLUGIN_PATH/enabled/*'
+++ basename /var/lib/dokku/core-plugins/enabled/docker-options
++ config-export docker-options
++ declare 'desc=Export plugin configuration'
++ declare plugin=docker-options
++ toml-export docker-options
++ /usr/local/bin/plugn ::: toml-export docker-options
++ cat /var/lib/dokku/core-plugins/config.toml
+ eval ''
+ [[ -x /var/lib/dokku/core-plugins/enabled/docker-options/install ]]
+ for plugin in '$PLUGIN_PATH/enabled/*'
+++ basename /var/lib/dokku/core-plugins/enabled/domains
++ config-export domains
++ declare 'desc=Export plugin configuration'
++ declare plugin=domains
++ toml-export domains
++ /usr/local/bin/plugn ::: toml-export domains
++ cat /var/lib/dokku/core-plugins/config.toml
+ eval ''
+ [[ -x /var/lib/dokku/core-plugins/enabled/domains/install ]]
+ /var/lib/dokku/core-plugins/enabled/domains/install
+ for plugin in '$PLUGIN_PATH/enabled/*'
+++ basename /var/lib/dokku/core-plugins/enabled/enter
++ config-export enter
++ declare 'desc=Export plugin configuration'
++ declare plugin=enter
++ toml-export enter
++ /usr/local/bin/plugn ::: toml-export enter
++ cat /var/lib/dokku/core-plugins/config.toml
+ eval ''
+ [[ -x /var/lib/dokku/core-plugins/enabled/enter/install ]]
+ for plugin in '$PLUGIN_PATH/enabled/*'
+++ basename /var/lib/dokku/core-plugins/enabled/git
++ config-export git
++ declare 'desc=Export plugin configuration'
++ declare plugin=git
++ toml-export git
++ /usr/local/bin/plugn ::: toml-export git
++ cat /var/lib/dokku/core-plugins/config.toml
+ eval ''
+ [[ -x /var/lib/dokku/core-plugins/enabled/git/install ]]
+ for plugin in '$PLUGIN_PATH/enabled/*'
+++ basename /var/lib/dokku/core-plugins/enabled/logs
++ config-export logs
++ declare 'desc=Export plugin configuration'
++ declare plugin=logs
++ toml-export logs
++ /usr/local/bin/plugn ::: toml-export logs
++ cat /var/lib/dokku/core-plugins/config.toml
+ eval ''
+ [[ -x /var/lib/dokku/core-plugins/enabled/logs/install ]]
+ for plugin in '$PLUGIN_PATH/enabled/*'
+++ basename /var/lib/dokku/core-plugins/enabled/named-containers
++ config-export named-containers
++ declare 'desc=Export plugin configuration'
++ declare plugin=named-containers
++ toml-export named-containers
++ /usr/local/bin/plugn ::: toml-export named-containers
++ cat /var/lib/dokku/core-plugins/config.toml
+ eval ''
+ [[ -x /var/lib/dokku/core-plugins/enabled/named-containers/install ]]
+ for plugin in '$PLUGIN_PATH/enabled/*'
+++ basename /var/lib/dokku/core-plugins/enabled/nginx-vhosts
++ config-export nginx-vhosts
++ declare 'desc=Export plugin configuration'
++ declare plugin=nginx-vhosts
++ toml-export nginx-vhosts
++ /usr/local/bin/plugn ::: toml-export nginx-vhosts
++ cat /var/lib/dokku/core-plugins/config.toml
+ eval ''
+ [[ -x /var/lib/dokku/core-plugins/enabled/nginx-vhosts/install ]]
+ /var/lib/dokku/core-plugins/enabled/nginx-vhosts/install
Adding user dokku to group adm
dpkg: error processing package dokku (--configure):
 subprocess installed post-installation script returned error exit status 1
Setting up herokuish (0.3.9) ...
Importing herokuish into docker (around 5 minutes)
Setting up libc-dev-bin (2.19-0ubuntu6.7) ...
Setting up linux-libc-dev:amd64 (3.13.0-83.127) ...
Setting up libc6-dev:amd64 (2.19-0ubuntu6.7) ...
Setting up manpages-dev (3.54-1ubuntu1) ...
Processing triggers for libc-bin (2.19-0ubuntu6.7) ...
Errors were encountered while processing:
 dokku
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

Additional info: Reason for NGINX 1.9.13 is for HTTP/2
```text
root@ubuntu-1gb-nyc3-01:~# nginx -v
nginx version: nginx/1.9.13
```

### LINKED ISSUE #2216: nginx-vhosts plugin should use HTTP/2 module in Nginx 1.9.5 and newer
When used with Nginx 1.9.5 or newer, the nginx-vhosts plugin writes a configration using the superseded “spdy” module:
```
listen      [::]:443 ssl spdy;
listen      443 ssl spdy;
```

This leads to a warning every time the configuration is reloaded:
```
invalid parameter "spdy": ngx_http_spdy_module was superseded by ngx_http_v2_module
in /home/dokku/…/nginx.conf:18
```

The replacement module is [ngx_http_v2_module](http://nginx.org/en/docs/http/ngx_http_v2_module.html).

Environment: Ubuntu 14.04.4 LTS, nginx 1.10.0, Dokku 0.5.7

### PR DESCRIPTION
**Pull request by josegonzalez**
Closes #2059.
Closes #2216.

### CODE REVIEW

File: plugins/nginx-vhosts/functions [hunk 1]
```diff
@@ -201,6 +228,10 @@ nginx_build_config() {
     xargs -i echo "-----> Configuring {}...(using $NGINX_TEMPLATE_SOURCE template)" <<< "$(echo "${SSL_VHOSTS}" "${NONSSL_VHOSTS}" | tr ' ' '\n' | sort -u)"
     # echo "sigil ${SIGIL_PARAMS[@]}"
     sigil "${SIGIL_PARAMS[@]}" > "$NGINX_CONF"
+    if [[ "$NGINX_TEMPLATE" == "$DEFAULT_NGINX_TEMPLATE" ]] && is_spdy_enabled "$(nginx -v 2>&1 | cut -d'/' -f 2)"; then
```
Comment (michaelshobbs) [inline plugins/nginx-vhosts/functions]: Leaving this to `sigil` would be cleaner than munging the output with `sed`

Comment (josegonzalez) [inline plugins/nginx-vhosts/functions]: So pass in `SPDY_SUPPORT` or something?

Comment (michaelshobbs) [inline plugins/nginx-vhosts/functions]: Exactly

Comment (josegonzalez) [inline plugins/nginx-vhosts/functions]: PRs to my PRs are welcome? :P

I'll take a look in a bit

**Comment by michaelshobbs** [comment]
don't merge this yet. i'm working on the sigil stuff

**Pull request merged**